mozregression/json_pushes.py

```diff
@@ -69,9 +69,9 @@ def pushes(self, **kwargs):
 
         response = retry_get(url)
         if response.status_code == 404:
-            raise MozRegressionError(
-                "The url %r returned a 404 error. Please check the"
-                " validity of the url." % url
+            raise EmptyPushlogError(
+                "The url %r returned a 404 error. The push may not be in"
+                " this repo (e.g., not merged yet)." % url
```

Comment:
👍 I think JsonPushes is the place to handle this. It looks like if we open this URL:

https://hg.mozilla.org/integration/mozilla-inbound/json-pushes?changeset=2f19e7b646e0a52fa855b75c868f0a3f3a990ad3&full=1

We get this raw json response:

```json
{
"error": "unknown revision '2f19e7b646e0a52fa855b75c868f0a3f3a990ad3'"
}
```

Do you think you could check the response data for a string of this kind before returning EmptyPushLogError? That way we'll fail correctly in case there is a 404 for some other reason (e.g. server misconfiguration)

Reply:
I'll do that. I think I still end up not needing MozRegressionError because raise_for_status() will do effectively the same thing if we 404 for another reason?

Reply:
Ah yes, you are right. You can remove the import from this file, and the linter should start passing again